### PR TITLE
Implement --debugger option to improve UX when debugging crashes

### PR DESCRIPTION
**Pull request by sfan5**
Instead of trying to direct users to enter the magic words to start gdb and then pull out a backtrace with this PR we can just tell them to add a single argument.
This is so seamless that you could even add it to your startup script and have e.g. the server always run under gdb.

## To do

This PR is a Work in Progress

- [x] proper Windows support (maybe [this](https://learn.microsoft.com/en-us/windows-server/administration/windows-commands/start) and [this](https://learn.microsoft.com/en-us/windows/win32/api/debugapi/nf-debugapi-isdebuggerpresent))

## How to test

`./bin/minetest --debugger`
optionally add a segfault bug somewhere first to see what happens

### CODE REVIEW

**Review by ericsomto — APPROVED** [review]

**Comment by celeron55** [comment]
I didn't test but I did read through the code and the code looks good.

**Review by celeron55 — APPROVED** [review]

File: src/porting.cpp [hunk 1]
```diff
@@ -729,6 +729,38 @@ void attachOrCreateConsole()
...
+	// and there isn't even a stdlib function to deal with it.
+	// Ref: https://learn.microsoft.com/archive/blogs/twistylittlepassagesallalike/everyone-quotes-command-line-arguments-the-wrong-way
+	if (!arg.empty() && arg.find_first_of(" \t\n\v\"") == std::string::npos)
+		return arg;
+
+	std::string ret;
+	ret.reserve(arg.size()+2);
+	ret.push_back('"');
+	for (auto it = arg.begin(); it != arg.end(); ++it) {
+		u32 back = 0;
```
Comment (TurkeyMcMac) [inline src/porting.cpp:745]: `back` appears to always be 0?

File: src/main.cpp [hunk 2]
```diff
@@ -341,6 +347,8 @@ static void set_allowed_options(OptionList *allowed_options)
 			_("Print even more information to console"))));
 	allowed_options->insert(std::make_pair("trace", ValueSpec(VALUETYPE_FLAG,
 			_("Print enormous amounts of information to log and console"))));
+	allowed_options->insert(std::make_pair("debugger", ValueSpec(VALUETYPE_FLAG,
+			_("Try to automatically attach a debugger before starting (convenience option)"))));
```
Comment (TurkeyMcMac) [inline src/main.cpp:351]:
```suggestion
			_("Try to automatically attach a debugger before starting (convenience option)"))));
```

**Review by TurkeyMcMac — COMMENTED** [review]
Works on Linux. I haven't tested thoroughly on Windows (I'm using Wine).

File: src/porting.cpp (hunk 1 above)
Comment (sfan5) [reply src/porting.cpp:745]: Uh oh, fixed.

**Review by TurkeyMcMac — APPROVED** [review]
Tested on Windows too now.

**Pull request merged**